packages/apps/tools/src/pages/faucet/new/index.tsx

```diff
@@ -87,58 +92,45 @@ const NewAccountFaucetPage: FC = () => {
     message?: string;
   }>({ status: 'idle' });
   const [pubKeys, setPubKeys] = useState<string[]>([]);
-  const [currentKey, setCurrentKey] = useState<string>('');
-  const [validRequestKey, setValidRequestKey] = useState<
-    FormFieldStatus | undefined
-  >();
 
   const { data: accountName } = useQuery({
     queryKey: ['accountName', pubKeys, chainID, pred],
-    queryFn: async () => {
-      if (pubKeys.length === 0) return '';
-      return await createPrincipal(pubKeys, chainID, pred);
-    },
-    initialData: '',
-  });
-
-  const schema = z.object({
-    name: z.string(),
-    pubKey: z.string().optional(),
+    queryFn: () => createPrincipal(pubKeys, chainID, pred),
```

Comment:
I added a third createPrincipal implementation in this PR: https://github.com/kadena-community/kadena.js/pull/1253/files#diff-64a159ddfe2566f68001bcacb9dbbb1e24e4bc1bb85c1e4794d6836b0fdbfafa. I do not think it is necessary to check for the signature in the tools implementations, since you are doing a dirtyRead for the transaction, but I may be wrong.